src/opm/output/eclipse/Summary.cpp

```diff
@@ -550,14 +553,72 @@ double efac( const std::vector<std::pair<std::string,double>>& eff_factors, cons
     return (it != eff_factors.end()) ? it->second : 1.0;
 }
 
-/*
-  This is bit dangerous, exactly how the ALQ value should be interpreted varies
-  between the different VFP tables. The code here assumes - without checking -
-  that it represents gas lift rate.
-*/
+inline quantity artificial_lift_quantity( const fn_args& args ) {
+    // Note: This function is intentionally supported only at the well level
+    // (meaning there's no loop over args.schedule_wells by intention).  Its
+    // purpose is to calculate WALQ only.
+    auto alq = quantity { 0.0, measure::identity };
+
+    if (args.schedule_wells.empty()) {
+        return alq;
+    }
+
+    const auto* well = args.schedule_wells.front();
```

Comment:
Should there be a `if (args.schedule_wells.size() > 1) throw()` here?

Reply:
> Should there be a `if (args.schedule_wells.size() > 1) throw()` here?

Not really.  An `assert()` **might** be appropriate, but if we ever get here with a well list of more than one item, then something's gone **horribly** wrong in [`find_wells()`](https://github.com/OPM/opm-common/blob/59efd8e98e5688fda64ca3efd66f32a433fb7c93/src/opm/output/eclipse/Summary.cpp#L2135-L2138) and we would see it the symptom in every `W*` summary vector.